tests/vector_tests.rs

```diff
@@ -149,3 +149,110 @@ fn blake2sp_vectors() {
     // count.
     assert_eq!(512, test_num);
 }
+
+fn blake2x_test<F: Fn(&[u8], &[u8], u64) -> Vec<u8>, F2: Fn(&[u8], u64, usize) -> Vec<u8>>(h0_hasher: F, b2_hasher: F2, variant_hash_length: usize, variant_name: &str) {
+    let mut test_num = 0u64;
+    for case in TEST_CASES.iter() {
+        if &case.hash == variant_name {
+            test_num += 1;
+
+            let input_bytes = hex::decode(&case.in_).unwrap();
+            let key = if !case.key.is_empty() {
+                hex::decode(&case.key).unwrap()
+            } else {
+                vec![]
+            };
+
+            let output_length = case.out.len()/2;
+            let encoded_output_length = (((output_length & ((1 << 8) - 1)) << 32) | ((output_length >> 8) << 40)) as u64;
+            let h0 = h0_hasher(&input_bytes, &key, encoded_output_length);
+
+            let num_hashes = (output_length + variant_hash_length - 1)/variant_hash_length;
```

Comment:
You might be able to avoid this clever arithmetic by using a while loop instead of a for loop. Something like:

    while buf.len() < output_length { ... }